fuel-p2p/Cargo.toml

```diff
@@ -14,15 +14,49 @@ description = "Fuel client networking"
 anyhow = "1.0"
 async-trait = "0.1"
 bincode = "1.3"
-fuel-core-interfaces = { path = "../fuel-core-interfaces", features = ["serde"], version = "0.14.1" }
+fuel-core-interfaces = { path = "../fuel-core-interfaces", features = [
+    "serde",
+], version = "0.14.1" }
 fuel-metrics = { path = "../fuel-metrics", version = "0.14.1" } # TODO make this a feature
 futures = "0.3"
 futures-timer = "3.0"
 ip_network = "0.4"
-libp2p = { version = "0.49", default-features = false, features = [
-    "dns-tokio", "gossipsub", "identify", "kad", "mdns-tokio", "mplex", "noise",
-    "ping", "request-response", "secp256k1", "tcp-tokio", "yamux", "websocket"
+libp2p = { version = "=0.50.0", default-features = false, features = [
+    "dns",
+    "gossipsub",
+    "identify",
+    "kad",
+    "macros",
+    "mdns",
+    "mplex",
+    "noise",
+    "ping",
+    "request-response",
+    "secp256k1",
+    "tcp",
+    "tokio",
+    "yamux",
+    "websocket",
 ] }
+libp2p-core = "=0.38.0"
+libp2p-dns = "=0.38.0"
+libp2p-gossipsub = "=0.43.0"
+libp2p-identify = "=0.41.0"
+libp2p-kad = "=0.42.0"
+libp2p-mdns = "=0.42.0"
+libp2p-metrics = "=0.11.0"
+libp2p-mplex = "=0.38.0"
+libp2p-noise = "=0.41.0"
+libp2p-ping = "=0.41.0"
+libp2p-quic = "=0.7.0-alpha"
+libp2p-request-response = "=0.23.0"
+libp2p-swarm = "=0.41.0"
+libp2p-swarm-derive = "=0.30.2"
```

Comment:
this version has been yanked: https://crates.io/crates/libp2p-swarm-derive/0.30.2

Reply:
as discussed over slack, let's keep the yanked version until libp2p 0.50 is patched to work with swarm-derive 0.31